commercetools.Sdk/commercetools.Base.Client/ExceptionFactory.cs

```diff
@@ -2,75 +2,90 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Net.Http;
+using System.Runtime.CompilerServices;
 using commercetools.Base.Client.Error;
+using commercetools.Base.Serialization;
 
 namespace commercetools.Base.Client
 {
     public class ExceptionFactory
     {
-        public static ApiHttpException Create(HttpRequestMessage request, HttpResponseMessage response)
+        public static ApiHttpException Create(HttpRequestMessage request, HttpResponseMessage response, Func<HttpResponseMessage, Object> errorResponseBodyMapper)
         {
-            return (int) response.StatusCode >= 500
-                ? CreateServerException(request, response)
-                : CreateClientException(request, response);
+            Object errorResponse = null;
+            string bodyDeserializationException = null;
+            
+            try
+            {
+                errorResponse = errorResponseBodyMapper(response);
+            }
+            catch (Exception deserializationException)
+            {
+                bodyDeserializationException = deserializationException.Message;
+            }
+            var createdException = (int) response.StatusCode >= 500
+                ? CreateServerException(request, response, errorResponse)
+                : CreateClientException(request, response, errorResponse);
+
+            createdException.BodyDeserializationException = bodyDeserializationException;
```

Comment:
I think this is not needed as it gives not much additional information. Should be fine to just silently ignore it.

Only think what has to be double checked is what happens if ResponseBody is null and someone uses the AsErrorResponse extension. Maybe the ResponseBody property has to be marked as Nullable

Reply:
I silently ignore it. Also if the responseBody is null and someone used AsErrorResponse extension, it will return null also. we don't need to mark ResponseBody property as Nullable as reference types is nullable by default